spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -109,6 +112,58 @@ object Engine {
   private var physicalCoreNumber = -1
   private var nodeNum: Int = -1
 
+  private var gatewayServer: py4j.GatewayServer = null
+
+
+  private[bigdl] def createJavaGateway(driverPort: Int): Unit = {
```

Comment:
should it be a singleton here? i.e createJavaGateway is invoked multiple times on the same Executor.

Reply:
On java side, Engine is already a singleton, so it won't create multiple servers.